client/asset/btc/walletclient.go

```diff
@@ -56,6 +70,54 @@ func newWalletClient(node rpcClient, segwit bool, chainParams *chaincfg.Params)
 // sent via RawRequest.
 type anylist []interface{}
 
+func (wc *walletClient) EstimateSmartFee(confTarget int64, mode *btcjson.EstimateSmartFeeMode) (*btcjson.EstimateSmartFeeResult, error) {
+	res := new(btcjson.EstimateSmartFeeResult)
+	return res, wc.call(methodEstimateSmartFee, anylist{confTarget, mode}, res)
+}
+
+func (wc *walletClient) SendRawTransaction(tx *wire.MsgTx, allowHighFees bool) (*chainhash.Hash, error) {
+	txBytes, err := serializeMsgTx(tx)
+	if err != nil {
+		return nil, err
+	}
+	return wc.callHashGetter(methodSendRawTransaction, anylist{
+		hex.EncodeToString(txBytes), allowHighFees})
```

Comment:
The trouble you're going to run into here is with how the sendrawtransaction RPC's syntax has changed.  See https://github.com/decred/dcrdex/pull/633

The `allowHighFees` boolean argument changed to a `maxFeeRate` numeric argument.  btcd's rpclient detects which to use depending on the version obtained from `getnetworkinfo`.  We'll need a switch to say which syntax to use.  e.g. https://github.com/decred/dcrdex/pull/633/files#diff-1b07e494f4bdd99b2a617ac110fbadc081e5b6ae1abf23f64032bbee5ebebef1R654

You are welcome to cherry-pick the two commits from that now-closed PR, or just make the same changes.  The btc_test.go changes might also be of help.

Reply:
🔥